backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/event/remote/RabbitMQEventBusRemoteEndpoint.java

```diff
@@ -84,178 +71,72 @@ public void setEventBusFactory(@NonNull final IEventBusFactory eventBusFactory)
 			RabbitMQEventBusConfiguration.DefaultQueueConfiguration.QUEUE_NAME_SPEL,
 			RabbitMQEventBusConfiguration.CacheInvalidationQueueConfiguration.QUEUE_NAME_SPEL,
 			RabbitMQEventBusConfiguration.AccountingQueueConfiguration.QUEUE_NAME_SPEL,
+			RabbitMQEventBusConfiguration.ManageSchedulerQueueConfiguration.QUEUE_NAME_SPEL,
+			RabbitMQEventBusConfiguration.AsyncBatchQueueConfiguration.QUEUE_NAME_SPEL,
+			RabbitMQEventBusConfiguration.MaterialEventsQueueConfiguration.QUEUE_NAME_SPEL
 	})
-	public void onRemoteEvent(
+	public void onEvent(
 			@Payload final Event event,
 			@Header(HEADER_SenderId) final String senderId,
-			@Header(HEADER_TopicName) final String topicName)
+			@Header(HEADER_TopicName) final String topicName,
+			@Header(HEADER_TopicType) final Type topicType)
 	{
-		final Topic topic = Topic.of(topicName, Type.REMOTE);
-		final IEventBus localEventBus = eventBusFactory.getEventBusIfExists(topic);
-		if (localEventBus == null)
+		final Topic topic = Topic.of(topicName, topicType);
+
+		if (topic.getType() != topicType)
 		{
-			logger.debug("onRemoteEvent - localEventBus for topicName={} is null; -> ignoring event", topicName);
+			logger.debug("onEvent - different local topicType for topicName:[{}], remote type:[{}] local type=[{}]; -> ignoring event", topicName, topicType, topic.getType());
 			return;
 		}
-		if (Objects.equals(getSenderId(), senderId))
+
+		final IEventBus localEventBus = eventBusFactory.getEventBus(topic);
+
+		if (Type.LOCAL == topic.getType() && !Objects.equals(getSenderId(), senderId))
 		{
-			logger.debug("onRemoteEvent - event's senderId is equal to the *local* sender id; -> ignoring event", senderId);
+			logger.debug("onEvent - type LOCAL but event's senderId={} is not equal to the *local* senderId={}; -> ignoring event", senderId, getSenderId());
```

Comment:
It might be as local events should only be posted to local queues, but I figured the guard doesn't do us any harm.
Should an error be thrown instead?